### PR TITLE
dev/productdisplay - Add entities to UI

### PR DESCRIPTION
**Pull request by AlanRobinson1337**

### CODE REVIEW

File: src/main/java/org/somerville/swag/data/entity/Order.java [hunk 1]
```diff
@@ -40,4 +40,9 @@ public void setCustomerId(int customerId) {
     public void setOrderLinesList(ArrayList<OrderLine> orderLinesList) {
         this.orderLinesList = orderLinesList;
     }
+
+    public ArrayList<OrderLine> addOrderLineToOrder(ArrayList<OrderLine> existingOrderLines, OrderLine orderLine){
```
Comment (sgoudham) [inline src/main/java/org/somerville/swag/data/entity/Order.java]: - Can reduce the complexity of this method by removing existingOrderLines parameter as you are already within Order.java. 
- Method name could also be named to "addOrderLine" for increased readability.

Good job overall!

File: src/main/java/org/somerville/swag/display/Basket.java [hunk 2]
```diff
@@ -12,6 +19,46 @@
...
+
+        String[] columNames = {"Product" , "Quantity"};
+        DefaultTableModel model = new DefaultTableModel(columNames,0) {
+            @Override
+            public boolean isCellEditable(int row, int column) {
+                return false;
+            }
+        };
+
+        for (OrderLine orderLine : lines) {
```
Comment (sgoudham) [inline src/main/java/org/somerville/swag/display/Basket.java:53]: Glad you went with the Enhanced For Loop which provides increased readability and maintainability while iterating through a List.

File: src/main/java/org/somerville/swag/display/LandingPage.java [hunk 3]
```diff
@@ -1,47 +1,96 @@
...
+        JSpinner spinner = new JSpinner();
+        JFormattedTextField spin = ((JSpinner.DefaultEditor) spinner.getEditor()).getTextField();
+        spin.setEditable(false);
+        quantitySpinner.setModel(model);
+
+        lblDesc.setText(product.getDescription());
+        lblProductTitle.setText(product.getName());
+
+        ImageIcon productImageIcon1 = new ImageIcon(product.getImagepath());
+        Image scaledProductImage1 = productImageIcon1.getImage().getScaledInstance(400, 350, Image.SCALE_SMOOTH);
```
Comment (RobertRanson) [inline src/main/java/org/somerville/swag/display/LandingPage.java:83]: Having set size for the images with smooth scaling is a good idea, as they are different sizes

File: src/main/java/org/somerville/swag/display/SignUp.java [hunk 4]
```diff
@@ -1,23 +1,60 @@
...
+                    !txtPhoneNo.getText().strip().matches("[0-9]{11}") && //only numbers,, 11 characters
+                    !String.valueOf(txtPassword.getPassword()).equals(String.valueOf(txtPasswordConfirm.getPassword())) //both password fields must match
+            )   //order of pressidence/logic for and and or inside the same ifs???
+            {
+                JOptionPane.showMessageDialog(root, "Something went wrong and the code is so bad I dont know what",
+                        "Sign In error", JOptionPane.ERROR_MESSAGE);
+                //regex expression for all txtFields
+                //include J option Panes as error messages
+                //database call once all validation is completed
+            }
```
Comment (RobertRanson) [inline src/main/java/org/somerville/swag/display/SignUp.java:54]: Amazing work using regex for text fields, smart way of doing it!

**Review by RobertRanson — APPROVED** [review]

**Review by hzl616 — APPROVED** [review]
Effective use of the method to get stock level

**Review by erinmaria98 — APPROVED** [review]

File: src/main/java/org/somerville/swag/display/Basket.java [hunk 5]
```diff
@@ -1,26 +1,81 @@
...
+        }
+        //End STUB
+
+
+        List<OrderLine> lines = o.getOrderLines(); //replace o.getOrderLines() with customer.getOrder().getOrderLinesAsList()
+
+        //---------------------TABLE LAYOUT-----------------------------------------------------------------------------
+
+        String[] columNames = {"Product" , "Quantity", "Price"};
+        DefaultTableModel model = new DefaultTableModel(columNames,0) {
```
Comment (2013884f) [inline src/main/java/org/somerville/swag/display/Basket.java:50]: columNames typo?

**Review by 2013884f — APPROVED** [review]
Good work on the redgex to validate card numbers with relevent error messages

File: src/main/java/org/somerville/swag/display/Basket.java (hunk 5 above)
Comment (AlanRobinson1337) [reply src/main/java/org/somerville/swag/display/Basket.java:50]: good catch! will refactor

**Pull request merged**